runner/ollamarunner/runner.go

```diff
@@ -883,23 +874,16 @@ func (s *Server) completion(w http.ResponseWriter, r *http.Request) {
 			return
 		case content, ok := <-seq.responses:
 			if ok {
-				if strings.TrimSpace(content) == lastToken {
-					tokenRepeat++
-				}
-				if tokenRepeat == tokenRepeatLimit {
-					http.Error(w, "token repeat limit reached", http.StatusInternalServerError)
+				var thinking string
+				var err error
+				content, thinking, err = parser.AddContent(content)
+				if err != nil {
+					fmt.Println("err", err)
+					http.Error(w, err.Error(), http.StatusInternalServerError)
 					seq.doneReason = llm.DoneReasonTokenRepeatLimit
```

Comment:
It's a little bit odd to set the `doneReason` to a specific reason for all errors. One possible solution is to check for a specific error value being returned and use that to set doneReason (or a generic one otherwise). Or if there will be a lot of reasons, just pass it back directly `AddContent`.